quakemigrate/io/data.py

```diff
@@ -309,12 +303,16 @@ def _load_from_path(self, starttime, endtime):
                                                  dtime=now)
                 files = chain(files, self.archive_path.glob(file_format))
             else:
-                for station in self.stations:
+                for n, s, l, c in zip(self.stations['Network'], 
+                                  self.stations['Station'],
+                                  self.stations['Location'],
+                                  self.stations['Channel']):
                     file_format = self.format.format(year=now.year,
                                                      month=now.month,
                                                      day=now.day,
                                                      jday=now.julday,
-                                                     station=station,
+                                                     network=n, station=s, 
+                                                     location=l, channel=c,
```

Comment:
Same as previous suggestion - better to just loop over the rows and extract each thing explicitly. There's absolutely no need to use single letters for variables and should be avoided as much as possible.